Iter24

Aleale16 opened the pull request.

Aleale16 committed:
modified:   go.work.sum
	modified:   internal/app/initconfig/initconfig.go
	modified:   internal/app/server/server.go

Aleale16 committed:
modified:   internal/app/server/server.go

Aleale16 committed:
modified:   internal/app/initconfig/initconfig.go
	modified:   internal/app/server/server.go

Aleale16 committed:
modified:   internal/app/server/server.go

Aleale16 committed:
modified:   internal/app/server/server.go

Aleale16 committed:
modified:   internal/app/initconfig/initconfig.go
	new file:   internal/app/initconfig/initconfigfromfile.go

Aleale16 committed:
modified:   cmd/shortener/main.go

Aleale16 committed:
modified:   cmd/shortener/main.go

Aleale16 committed:
modified:   internal/app/initconfig/initconfigfromfile.go

Aleale16 committed:
modified:   internal/app/initconfig/initconfigfromfile.go

Aleale16 committed:
modified:   internal/app/handler/handler.go
	modified:   internal/app/initconfig/initconfig.go
	modified:   internal/app/initconfig/initconfigfromfile.go
	modified:   internal/app/server/server.go

Aleale16 committed:
modified:   go.mod
	modified:   go.sum
	modified:   go.work
	modified:   go.work.sum
	new file:   internal/app/grpcapp/go.mod
	new file:   internal/app/grpcapp/go.sum
	new file:   internal/app/grpcapp/proto/grpcapp.pb.go
	new file:   internal/app/grpcapp/proto/grpcapp.proto
	new file:   internal/app/grpcapp/proto/grpcapp_grpc.pb.go
	new file:   internal/app/grpcserver/go.mod
	new file:   internal/app/grpcserver/go.sum
	new file:   internal/app/grpcserver/grpcserver.go
	new file:   internal/app/grpcserver/main.go

Bazys commented on cmd/shortener/main.go:
```diff
@@ -62,8 +65,15 @@ func main() {
+	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGTERM, syscall.SIGINT, syscall.SIGQUIT)
+	defer stop()
 
+	go server.Start()
```
а как же насчет корректно остановить сервер?
ты в него `server.Start()` контекст не передаешь и он не знает о том что приложение завершается и не запишет те данные которые в памяти на диск

Aleale16 committed:
modified:   cmd/shortener/main.go
	modified:   internal/app/initconfig/initconfig.go
	modified:   internal/app/server/server.go
	new file:   internal/app/server/serverhttps.go

Aleale16 committed:
modified:   internal/app/server/server.go

Aleale16 committed:
modified:   internal/app/server/server.go

Aleale16 committed:
modified:   internal/app/server/server.go

Aleale16 committed:
modified:   internal/app/server/server.go

Aleale16 committed:
modified:   internal/app/server/server.go

Bazys commented on internal/app/grpcapp/go.sum:
```diff
@@ -0,0 +1,20 @@
+github.com/golang/protobuf v1.5.0/go.mod h1:FsONVRAS9T7sI+LIUmWTfcYkHO4aIWwzhcaSAoJOfIk=
```
заачем go.mod в каждой папке?
у ьебя это все равно лежит в internal никто снаружи этого пакета, эти файлы не увидит

Bazys commented on internal/app/grpcserver/main.go:
```diff
@@ -0,0 +1,30 @@
+	"google.golang.org/grpc"
+)
+
+func main() {
```
а зачем нам тут main
разве не можем в основной точке входа его поднимать?
не должно быть два бинарника

Aleale16 committed:
modified:   cmd/shortener/main.go
	modified:   go.mod
	modified:   go.sum
	modified:   go.work
	modified:   go.work.sum
	deleted:    internal/app/grpcapp/go.mod
	deleted:    internal/app/grpcapp/go.sum
	new file:   internal/app/grpcapp/grpcapp.go
	new file:   internal/app/grpcapp/grpcserverstart.go
	modified:   internal/app/grpcapp/proto/grpcapp.pb.go
	modified:   internal/app/grpcapp/proto/grpcapp.proto
	modified:   internal/app/grpcapp/proto/grpcapp_grpc.pb.go
	deleted:    internal/app/grpcserver/go.mod
	deleted:    internal/app/grpcserver/go.sum
	deleted:    internal/app/grpcserver/grpcserver.go
	deleted:    internal/app/grpcserver/main.go

Aleale16 committed:
Merge branch 'iter24' of https://github.com/Aleale16/practicumDev into iter24

Aleale16 committed:
modified:   internal/app/grpcapp/grpcapp.go
	modified:   internal/app/grpcapp/proto/grpcapp.pb.go
	modified:   internal/app/grpcapp/proto/grpcapp.proto
	modified:   internal/app/storage/storage.go

Aleale16 committed:
modified:   internal/app/grpcapp/grpcapp.go

Aleale16 committed:
modified:   internal/app/grpcapp/grpcapp.go
	modified:   internal/app/grpcapp/grpcserverstart.go
	renamed:    internal/app/grpcapp/proto/grpcapp.pb.go -> internal/app/proto/grpcapp.pb.go
	renamed:    internal/app/grpcapp/proto/grpcapp.proto -> internal/app/proto/grpcapp.proto
	renamed:    internal/app/grpcapp/proto/grpcapp_grpc.pb.go -> internal/app/proto/grpcapp_grpc.pb.go

Aleale16 committed:
modified:   cmd/shortener/main.go
	modified:   internal/app/grpcapp/grpcapp.go
	modified:   internal/app/grpcapp/grpcserverstart.go

Aleale16 committed:
modified:   cmd/shortener/main.go
	modified:   internal/app/grpcapp/grpcserverstart.go

Aleale16 committed:
modified:   cmd/shortener/main.go

Aleale16 committed:
modified:   cmd/shortener/main.go
	modified:   internal/app/grpcapp/grpcserverstart.go

Aleale16 committed:
modified:   cmd/shortener/main.go

Aleale16 committed:
modified:   cmd/shortener/main.go
	modified:   internal/app/server/server.go

Aleale16 committed:
modified:   cmd/shortener/main.go
